test/system/src/test/java/io/pravega/test/system/SegmentReaderAPITest.java

```diff
@@ -0,0 +1,510 @@
+
+        ImmutableMap<Segment, Long> startStreamCut1 = ImmutableMap.of(segment3, 0L);
+        Map<Stream, StreamCut> startSC1 = ImmutableMap.of(Stream.of(streamScope, streamName),
+                new StreamCutImpl(Stream.of(streamScope, streamName), startStreamCut1));
+        log.info("startSC1 :: {} ", startSC1);
+
+        reader0.close();
+        ReaderGroupConfig readerGroupConfig2 = ReaderGroupConfig.builder()
+                .disableAutomaticCheckpoints()
+                .startingStreamCuts(startSC1)
+                .build();
+        readerGroup.resetReaderGroup(readerGroupConfig2);
+
+        //After scale down reading the event
+        reader0 = clientFactory.createReader(readerName,
+                readerGroupName,
+                new UTF8StringSerializer(),
+                ReaderConfig.builder().build());
+        assertEquals(readEvent(reader0), 5);
+        reader0.close();
+
+        ArrayList<SegmentRange> rangeList3 = Lists.newArrayList(batchClient.getSegments(stream, StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+        Map<Segment, Long> map = rangeList3.stream().collect(
+                Collectors.toMap(SegmentRange::getSegment, SegmentRange::getEndOffset));
+
+        // Getting streamcut at the tail of segment 1 and segment 2
+        StreamCut streamCut = new StreamCutImpl(stream,
+                ImmutableMap.of(segment1, map.get(segment1), segment2, map.get(segment2)));
+
+        StreamCut nextStreamCut3 = batchClient.getNextStreamCut(streamCut, 180L);
+        log.info("Next stream cut3 {}", nextStreamCut3);
+
+        assertTrue(nextStreamCut3 != null);
+        assertTrue(nextStreamCut3.asImpl().getPositions().size() == 1);
+        assertTrue(150L <= nextStreamCut3.asImpl().getPositions().get(segment3).longValue());
+    }
+
+    private void writeEvents(int numberOfEvents, EventStreamWriter<String> writer) {
+        Supplier<String> routingKeyGenerator = () -> String.valueOf(random.nextInt());
+        IntStream.range(0, numberOfEvents).forEach(v -> writer.writeEvent(routingKeyGenerator.get(), DATA_OF_SIZE_30).join());
+    }
+
+    private static ReaderGroupConfig getReaderGroupConfig(StreamCut startStreamCut, StreamCut endStreamCut, Stream stream) {
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder()
+                .disableAutomaticCheckpoints()
+                .stream(stream, startStreamCut, endStreamCut)
+                .build();
+        return readerGroupConfig;
+    }
+
+    private static int readEvent(EventStreamReader<String> reader) {
+        int readCount = 0;
+        EventRead<String> event = null;
+        do {
+            event = reader.readNextEvent(10000);
+            log.debug("Read event: {}.", event.getEvent());
+            if (event.getEvent() != null) {
+                readCount++;
+            }
+            //Reading until all the written events are read, else the test will timeout.
```

Comment:
This is going to add a 10 second delay to the test. If the number of events that are expected to be read is passed in as an argument it could exit when readCount reaches that number.

Reply:
added expectedReadCount parameter to the method.